sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorConnection.java

```diff
@@ -350,7 +350,7 @@ private synchronized Connection getOrCreateConnection() throws IOException {
             connection = reactor.connectionToHost(handler.getHostname(), handler.getProtocolPort(), handler);
 
             reactorExceptionHandler = new ReactorExceptionHandler();
-            executor = new ReactorExecutor(reactor, Schedulers.single(), connectionId,
+            executor = new ReactorExecutor(reactor, Schedulers.newSingle("new-reactor-connection"), connectionId,
```

Comment:
You are in charge of disposing of this Scheduler as well since we instantiated it. Do we take care of that?